UDS-1773: feat(unity-bootstrap-theme): changed h4 to h2 and added styles for this change

juanmitriatti opened the pull request:
feat(unity-bootstrap-theme): changed h4 to h2 and added styles for this change

UDS-1773

### Description

<!-- Description of problem -->
<!-- Solution -->
<!-- Testing Steps -->

### Links

- [JIRA ticket](https://asudev.jira.com/browse/UDS-1773)
- [Unity reference site](https://asu.github.io/asu-unity-stack/)
- [Unity Design Kit](https://shared-assets.adobe.com/link/fb14b288-bf63-47e0-5d30-384de5560455)

### FOR APPROVERS

- [Percy build approval](https://percy.io/5eae92d9/-all-UDS-packages)

juanmitriatti committed:
feat(unity-bootstrap-theme): changed h4 to h2 and added styles for this change

UDS-1773

mlsamuelson reviewed (changes requested):
This looks good. I can see from the scss changes that this will be backwards compatible so this is good as-is.  In the future if you ever have a PR that updates a `unity-bootstrap-theme` component in a breaking way - usually it means the markup has changed and won't render correctly until templates are updated by people using the component - then you will want to make your commit message as a "breaking change."

There is also a version of the anchor menu in the `components-core` package, and this update should get made there, too. When we update one, we want to update the other. Can you add that to this PR? When you make the commit for those changes, you'll want to use `components-core` as the scope in the commit message so it gets an updated release.

juanmitriatti committed:
fix(components-core): switch h4 to h2 on AnchorMenu

BREAKING CHANGE: It is necessary to replicate the changes from the unity-bootstrap-theme in the
component-core.

UDS-1773

mlsamuelson commented:
Thank you for the update. In both place, @juanmitriatti . Looks good.
